### PR TITLE
snmpcheck: init at 1.9

### PR DESCRIPTION
**Pull request by elohmeier**
###### Motivation for this change
#81418

###### Things done
Added snmpcheck package.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/snmpcheck/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    postBuild = ''
+      ln -sf ${ruby}/bin/* $out/bin
+    '';
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "snmpcheck";
+  version = "1.9";
+  src = fetchurl {
+    url = "http://www.nothink.org/codes/snmpcheck/snmpcheck-${version}.rb";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snmpcheck/default.nix:16]:
```suggestion
    url = "https://www.nothink.org/codes/snmpcheck/snmpcheck-${version}.rb";
```

File: pkgs/tools/networking/snmpcheck/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    cp $src $out/bin/snmp-check
+
+    substituteInPlace "$out/bin/snmp-check" \
+      --replace "#!/usr/bin/env ruby" "#!${rubyEnv}/bin/ruby"
+    chmod +x $out/bin/snmp-check
+  '';
+
+  meta = with lib; {
+    description = "SNMP enumerator";
+    homepage = "http://www.nothink.org/codes/snmpcheck/";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snmpcheck/default.nix:33]:
```suggestion
    homepage = "https://www.nothink.org/codes/snmpcheck/";
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127655` at c7bb4003 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>snmpcheck</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/networking/snmpcheck/default.nix:22:3:
```
   |
22 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/networking/snmpcheck/default.nix (hunk 2 above)
Comment (elohmeier) [reply pkgs/tools/networking/snmpcheck/default.nix]: Thanks, updated.

**Comment by elohmeier** [comment]
Updated the package to use rubyEnv.wrappedRuby

**Review by SuperSandro2000 — COMMENTED** [review]
I think it would be easier if we package it like this https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/ruby.section.md#adding-a-gem-to-the-default-gemset-adding-a-gem-to-the-default-gemset

**Comment by elohmeier** [comment]
> I think it would be easier if we package it like this https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/ruby.section.md#adding-a-gem-to-the-default-gemset-adding-a-gem-to-the-default-gemset

@SuperSandro2000 thanks for the tip, that worked well. I have updated the PR accordingly.

**Pull request merged**